### ISSUE TITLE
Alert.Link Has An Incorrect ARIA Role

### ISSUE THREAD

**Issue by LeftyDaniels** [react-bootstrap/react-bootstrap#5294]
## Describe the bug

The Alert.Link component gives the \<a\> element a role of “button”, which is inaccurate because it doesn't have the functionality of a \<button\> element.

## To Reproduce

1. Go to https://react-bootstrap.github.io/components/alerts/#links
2. Inspect any of the <a> elements in the Alert component examples
3. Observe the HTML attribute [role="button"]

## Expected behavior

According to MDN:

> "The button role identifies an element as a button to screen readers. A button is a widget used to perform actions such as submitting a form, opening a dialog, cancelling an action, or performing a command such as inserting a new record or displaying information."

No role attribute is required here because the correct HTML element is being used.

## Environment (please complete the following information)

- Operating System: macOS
- Browser, Version: Microsoft Edge 83.0.478.61 (64-bit)
- React-Bootstrap Version: 1.2.2

**Comment by taion** [comment]
Okay, that's weird. Do you want to send in a PR to fix?

**Comment by LeftyDaniels** [comment]
Sure, I can do that.

**Comment by LeftyDaniels** [comment]
Ok, so I looked further into it and the Alert.Link component is just a renamed SafeAnchor component. In the SafeAnchor component there is logic to determine if the [href] attribute is either empty or equal to the "#" character and if it is then add the [role="button"] attribute.

There is a comment that says that the cursor will have incorrect styling if there isn't an [href] attribute and the added [role="button"] attribute is to correct the styling. I looked to see what styles are being added and it is adding a "cursor: pointer" style to the \<a\> element. It is true that the \<a\> element will not have that style if there is no [href] attribute present, at least in Microsoft Edge Chromium and Mozilla Firefox.

Is it the desire to change the browser's default behavior so that the "cursor: pointer" style is always present?

If it isn't, then we can just remove this logic and that will resolve the issue.

If it is, then I recommend adding that style to the .alert-link class or adding an [href="#"] attribute to the React-Bootstrap SafeAnchor component in place of the [role="button"] attribute.

**Comment by taion** [comment]
I don't think we want to set up an `href` that isn't what the user defined there. It actually does make sense to show `href`-less links as "buttons" in some sense.

Maybe we can just fix the example to change the `<Alert.Link>`s to have an explicit `href="#"`?

**Comment by LeftyDaniels** [comment]
I can understand that. Don't add one if they didn't specify one.

I'll change the logic to only add `[role="button"]` if the `href` prop is empty.

I will also add that there are some further considerations for adding the `[role="button]` attribute and MDN has a great article covering the use cases and approaches for using it.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by LeftyDaniels** [comment]
Created a pull request for this that is a little different from our conversation here.

There was already logic in place to add an `href` if it was empty, so I just removed the `role` attribute.

#5296
